README.md

```diff
@@ -25,162 +25,90 @@ implementation "com.careem.mockingbird:mockingbird:$mockingBirdVersion"
-        const val value = "value"
-        const val value1 = "value1"
-        const val value2 = "value2"
-        const val object1 = "object1"
-    }
-
-    override fun method1(str: String) = mockUnit(
-        methodName = Method.method1,
-        arguments = mapOf(
-            Arg.str to str
-        )
-    )
-
-    override fun method2(str: String, value: Int) = mockUnit(
-        methodName = Method.method2,
-        arguments = mapOf(
-            Arg.str to str,
-            Arg.value to value
-        )
-    )
-
-    override fun method3(value1: Int, value2: Int): Int = mock(
-        methodName = Method.method3,
-        arguments = mapOf(
-            Arg.value1 to value1,
-            Arg.value2 to value2
-        )
-    )
-
-    override fun method4(object1: Object): Int = mock(
-        methodName = Method.method4,
-        arguments = mapOf(
-            Arg.object1 to object1
-        )
-    )
-}    
+classpath("com.careem.mockingbird:mockingbird-compiler:2.0.0-beta05")
 ```
 
-### Spies
-
-When you need a combination of real behavior and mocked behavior you can use `spy` with spy you wrap wrap a real
-implementation. Doing so Mocking Bird will record the interactions with the spied object.
-
-To mock a specific invocation you can use the spied object like a normal mock, see sections below for further details.
-
-A Spy sample object is reported here
+be also sure you have `mavenCentral` in your `buildscripts` repositories, your project `build.gradle.kts` might look
+similar the one below
 
 ```kotlin
-interface MyDependency {
-    fun method1(str: String)
-    fun method2(str: String, value: Int)
-    fun method3(value1: Int, value2: Int): Int
-    fun method4(): Int
-}
-
-class MyDependencySpy(private val delegate: MyDependency) : MyDependency, Spy {
-
-    object Method {
-        const val method1 = "method1"
-        const val method2 = "method2"
-        const val method3 = "method3"
-        const val method4 = "method4"
+buildscript {
+    repositories {
+        ...
+        mavenCentral()
     }
-
-    object Arg {
-        const val str = "str"
-        const val value = "value"
-        const val value1 = "value1"
-        const val value2 = "value2"
+    dependencies {
+        ...
+        classpath("com.careem.mockingbird:mockingbird-compiler:2.0.0-beta05")
     }
+}
+```
 
-    override fun method1(str: String) = spy(
-        methodName = Method.method1,
-        arguments = mapOf(
-            Arg.str to str
-        ),
-        delegate = { delegate.method1(str) }
-    )
+To generate mocks for a specific module you have first to apply the plugin in your module `build.gradle.kts`
 
-    override fun method2(str: String, value: Int) = spy(
-        methodName = Method.method2,
-        arguments = mapOf(
-            Arg.str to str,
-            Arg.value to value
-        ),
-        delegate = { delegate.method2(str, value) }
-    )
+```kotlin
+apply(plugin = "com.careem.mockingbird")
```

Comment:
can you move this under the kotlin section? and also put the groovy script since this is only for kotlin dsl

Reply:
Is ti worth tho? I specified Colton vs groovy only for the part where you specify the classes because it is different but for apply plugin I think we do not need to explain both no?